light/client.go

```diff
@@ -1146,3 +1148,28 @@ func (c *Client) providerShouldBeRemoved(err error) bool {
 		errors.As(err, &provider.ErrBadLightBlock{}) ||
 		errors.Is(err, provider.ErrConnectionClosed)
 }
+
+func (c *Client) Status(ctx context.Context) (*coretypes.LightClientInfo, error) { // return only light client info
+
+	chunks := make([]string, len(c.witnesses))
+	// If primary is in witness list we do not want to count it twice in the number of peers
+	primaryNotInWitnessList := 1
+	for i, val := range c.witnesses {
+		chunks[i] = val.String()
+		if chunks[i] == c.primary.String() {
+			primaryNotInWitnessList = 0
+		}
+	}
+
+	lastTrustedHeight, err := c.LastTrustedHeight()
+	lightClientInfo := coretypes.LightClientInfo{
+		Primary:                c.primary.String(),
+		Witnesses:              chunks,
+		NumPeers:               len(chunks) + primaryNotInWitnessList,
+		LastTrustedHeight:      lastTrustedHeight,
+		LastTrustedBlockHeight: c.latestTrustedBlock.Height,
```

Comment:
Ideally not, except one is read from the store, the other is part of the state. Does it every happen that those two are out of sync as in, the height has not been persisted yet and there is a failure inbetween? I will take a look at the code as well to verify this.

Reply:
I don't think it's perfectly atomic but good enough for this situation

Reply:
Will remove on of them then.

Reply:
Indeed they're not atomic since we persist the block in the store before we update the state. For instance he recovery/replay logic explicitly covers this case: https://github.com/tendermint/tendermint/blob/master/internal/consensus/replay.go#L401 

But for client purposes it's probably fine to just show one